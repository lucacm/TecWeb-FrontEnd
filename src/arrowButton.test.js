import React, { Component } from'react';
import PropTypes from "prop-types";
import { render, cleanup, fireEvent } from '@testing-library/react'
import Groups from "./screens/groups";
import Players from "./screens/players";
import LiveFixtures from "./screens/liveFixtures";
import { shallow } from 'enzyme';
import {ReactLogo} from 'react-share';
import { getAllByAltText, getByDisplayValue, getByText } from '@testing-library/react';


// CÃ³digo para resolver o conflito do useLocation (mude o location para o url do seu componente)
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useLocation: () => ({
      location: "localhost:3000/match"

    })
  }));

afterEach(cleanup)

it('Check if Arrow button was created', () => {
    const app = shallow(<Groups/>)
    expect(app.find('arrow').exists()).toMatchSnapshot()
});

it('Check if Arrow button was created', () => {
    const app = shallow(<Players/>)
    expect(app.find('arrow').exists()).toMatchSnapshot()
});

it('Check if Arrow button was created', () => {
    const app = shallow(<LiveFixtures/>)
    expect(app.find('arrow').exists()).toMatchSnapshot()
});


// describe('Arrow Button', () => {
    
//     const initialProps = {
//       id: '1',
//       awayTeam: 'Chealse',
//       homeTeam: 'Juventus',
//       score: '3-1',
//       date: '25-11-2020',
//       awayId:'2',
//       homeId:'3',
//       idUser:'4'
//     };
//     const container = shallow(<Match {...initialProps} />);
  
//     it('Check if Facebook share button was created', () => {
//       expect(container.find(FacebookShareButton).exists()).toBeDefined()
//     });
  
//     it('Check if Twitter share button was created', () => {
//         expect(container.find(TwitterShareButton).exists()).toBeDefined()
//       });
//     it('Check if Whatsapp share button was created', () => {
//     expect(container.find(WhatsappShareButton).exists()).toBeDefined()
//     });
//     it('Check if E-mail share button was created', () => {
//       expect(container.find(EmailShareButton).exists()).toBeDefined()
//     });
//     it('Check if Linkedin share button was created', () => {
//       expect(container.find(LinkedinShareButton).exists()).toBeDefined()
//     });

    
//   }); 
