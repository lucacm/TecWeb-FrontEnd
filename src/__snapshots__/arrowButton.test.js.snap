// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Arrow button was created 1`] = `false`;

exports[`Check if Arrow button was created 2`] = `false`;

exports[`Check if Arrow button was created 3`] = `false`;
